numpy/core/fromnumeric.py

```diff
@@ -63,10 +82,46 @@ def _wrapfunc(obj, method, *args, **kwds):
     # in its class, but its signature is not identical to that
     # of NumPy's. This situation has occurred in the case of
     # a downstream library like 'pandas'.
-    except (AttributeError, TypeError):
+    except (AttributeError, TypeError) as e:
```

Comment:
An alternative is to abstract the other way i.e. check the `AttributeError` in a separate method and the `TypeError` in a separate method e.. `_is_compat_attr_error` and `_is_compat_type_error`.  How about that?